lib/db.js

```diff
@@ -302,3 +311,59 @@ module.exports.anonymizeConferences = async () => {
     throw new Error("Error during conferences anonymization")
   }
 }
+
+// fetch conferences with email but no hashedEmail
+module.exports.getConferencesWithNoHashedEmails = async () =>
+   knex("conferences").whereNotNull("email").whereNull("hashedEmail").select("*")
+
+
+module.exports.updateHashedEmail = async ({ id, email }) =>
+   knex("conferences")
+    .where("id", id)
+    .update({
+      hashedEmail: format.hashForLogs(email),
+    })
+
+module.exports.getEmailsForSurvey = async () => {
+  try {
+    return knex("conferences")
+      .select(["conferences.email", "conferences.hashedEmail"])
+      .whereNotNull("email")
+      // We ignore deliberately old conferences which haven't hashedEmail
+      .whereNotNull("hashedEmail")
+      .whereRaw(`"expiresAt" < now()`)
+      .whereRaw(`"expiresAt" >= now() - interval '1 day'`)
+      // Don't send another email for users who have already be requested last week
+      .whereNotExists(function () {
+        this.select("*")
+          .from("users")
+          .whereRaw(`conferences."hashedEmail" = users."hashedEmail"`)
+          .whereRaw(`users."surveySentAt" >= now() - interval '1 week'`)
+      })
+      .groupBy(["conferences.email", "conferences.hashedEmail"])
+
+  } catch (err) {
+    console.error("Error during conferences anonymization", err)
+    throw new Error("Error during conferences anonymization")
+  }
+}
+
+const getUserFromHashedEmail = async (hashedEmail) => knex("users").select("*").where("hashedEmail", hashedEmail)
+
+module.exports.fillSurveyDateConference = async (hashedEmail) => {
+  try {
+    const [user] = await getUserFromHashedEmail(hashedEmail)
+
+    const surveySentAt = knex.fn.now()
+
+    if (!user) {
+      await knex("users").insert({ hashedEmail, surveySentAt }).returning("*")
+    } else {
+      await knex("users").where("id", user.id).update({ surveySentAt }).returning("*")
+    }
+  } catch (err) {
+    console.error(`Impossible de mettre à jour la conférence ${id}`, err)
```

Comment:
je trouve ca pratique de faire des messages uniques à chaque fonction, pour les retrouver facilement dans les logs (genre mettre le nom de la fonction dans le message)
Sauf si entre temps on a des stack traces correctes ? 
Peut etre que je chipote

Reply:
Je crois que de mon côté, j'enlèverais tous les try catch de db.js et je laisserai l'appelant se débrouiller avec l'erreur ;). 
Parce que là on ne fait qu'un console.error et on renvoie l'erreur, il n'y a pas de plus value. Mais par souci d'homogénéité, je le fais.